zzzz_modular_occulus/code/modules/projectiles/gun.dm

```diff
@@ -37,4 +40,73 @@
 			data["attachments"] += list(list("name" = A.name, "icon" = getAtomCacheFilename(A)))
 
 
-	return data
\ No newline at end of file
+	return data
+
+/obj/item/weapon/gun/Fire(atom/target, mob/living/user, clickparams, pointblank=0, reflex=0)
+	if(!user || !target) return
+
+	if(world.time < next_fire_time)
+		if (!suppress_delay_warning && world.time % 3) //to prevent spam
+			to_chat(user, SPAN_WARNING("[src] is not ready to fire again!"))
+		return
+
+
+	add_fingerprint(user)
+
+	if(!special_check(user))
+		return
+
+	var/shoot_time = (burst - 1)* burst_delay
+	user.setClickCooldown(shoot_time) //no clicking on things while shooting
+	next_fire_time = world.time + shoot_time
+
+	//actually attempt to shoot
+	var/turf/targloc = get_turf(target) //cache this in case target gets deleted during shooting, e.g. if it was a securitron that got destroyed.
+	for(var/i in 1 to burst)
+		var/obj/projectile = consume_next_projectile(user)
+		if(!projectile)
+			handle_click_empty(user)
+			break
+
+		projectile.multiply_projectile_damage(damage_multiplier)
+
+		projectile.multiply_projectile_penetration(penetration_multiplier)
+
+		projectile.multiply_pierce_penetration(pierce_multiplier)
+
+		projectile.multiply_projectile_step_delay(proj_step_multiplier)
+
+		projectile.change_knockback(knockback_change)
```

Comment:
Is this code any different from existing projectile code with the exception of adding the call to change_knockback()?

Reply:
This is functionally the same as the way existing code works.

However I am going to add a sanity check on this that isn't needed on other code.